[Bug]: LS stops working when the observe package is missing in the local cache

### Description

When a fresh package (No Dependencies.toml) is built with U13 and `observabilityIncluded=true` is set, the LS stops working. This is because the distribution has `ballerina/observe` 1.6.0 version but, one of the dependencies (`ballerinai/observe`) requires 1.5.0.

### Steps to Reproduce

1. Create a new package with observabilityIncluded=true
2. Open on VSCode
3. A pop-up open on the right hand side asking to pull the modules
4. Pulling results in the following error:

```
[Info  - 11:30:50 AM] Started pulling modules for project: /Users/asmaj/ballerina/testing/U13/hello
[Error - 11:30:53 AM] Pull modules failed for project: /Users/asmaj/ballerina/testing/U13/hello {, error: 'org.ballerinalang.compiler.BLangCompilerException: failed to load the module 'ballerinai/observe:0.0.0' from its BIR due to: cannot resolve module ballerina/observe:1.5.0'}
java.util.concurrent.CompletionException: org.ballerinalang.compiler.BLangCompilerException: failed to load the module 'ballerinai/observe:0.0.0' from its BIR due to: cannot resolve module ballerina/observe:1.5.0
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:791)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
Caused by: org.ballerinalang.compiler.BLangCompilerException: failed to load the module 'ballerinai/observe:0.0.0' from its BIR due to: cannot resolve module ballerina/observe:1.5.0
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.definePackage(BIRPackageSymbolEnter.java:234)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.definePackage(BIRPackageSymbolEnter.java:213)
	at io.ballerina.projects.ModuleContext.loadPackageSymbolInternal(ModuleContext.java:534)
	at io.ballerina.projects.ModuleCompilationState$8.compile(ModuleCompilationState.java:232)
	at io.ballerina.projects.ModuleContext.compile(ModuleContext.java:355)
	at io.ballerina.projects.PackageCompilation.compileModulesInternal(PackageCompilation.java:201)
	at io.ballerina.projects.PackageCompilation.compileModules(PackageCompilation.java:185)
	at io.ballerina.projects.PackageCompilation.compile(PackageCompilation.java:92)
	at io.ballerina.projects.PackageCompilation.from(PackageCompilation.java:87)
	at io.ballerina.projects.PackageContext.getPackageCompilation(PackageContext.java:249)
	at io.ballerina.projects.Package.getCompilation(Package.java:157)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.waitAndGetPackageCompilation(BallerinaWorkspaceManager.java:381)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.waitAndGetPackageCompilation(BallerinaWorkspaceManager.java:408)
	at org.ballerinalang.langserver.command.executors.PullModuleExecutor.lambda$resolveModules$8(PullModuleExecutor.java:152)
	at java.base/java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:787)
	... 6 more
Caused by: org.ballerinalang.compiler.BLangCompilerException: cannot resolve module ballerina/observe:1.5.0
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.defineImportPackage(BIRPackageSymbolEnter.java:423)
	at org.wso2.ballerinalang.util.LambdaExceptionUtils.lambda$rethrow$0(LambdaExceptionUtils.java:48)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.defineSymbols(BIRPackageSymbolEnter.java:408)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.definePackage(BIRPackageSymbolEnter.java:275)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.definePackage(BIRPackageSymbolEnter.java:256)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.definePackage(BIRPackageSymbolEnter.java:229)
	... 20 more

```


### Affected Version(s)

_No response_

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

--- a/gradle.properties
+++ b/gradle.properties
@@ -14,9 +14,9 @@
 
 org.gradle.caching=true
 group=io.ballerina
-version=1.5.1-SNAPSHOT
+version=1.6.0-SNAPSHOT
 
-ballerinaLangVersion=2201.12.0
+ballerinaLangVersion=2201.13.0
 spotbugsPluginVersion=6.0.18
 shadowJarPluginVersion=8.1.1
 downloadPluginVersion=5.4.0
@@ -28,7 +28,7 @@ openTelemetryVersion=1.7.0
 nettyCodecVersion=4.1.118.Final
 gsonVersion=2.10.1
 
-observeVersion=1.5.0
+observeVersion=1.6.0
 
 # Test Dependency Versions
 testngVersion=7.6.1

